src/routes/notifications/v1/notifications.controller.spec.ts

```diff
@@ -52,56 +60,110 @@ describe('Notifications Controller (Unit)', () => {
       .useModule(TestQueuesApiModule)
       .overrideModule(PostgresDatabaseModuleV2)
       .useModule(TestPostgresDatabaseModuleV2)
+      .overrideModule(NotificationsRepositoryV2Module)
+      .useModule(TestNotificationsRepositoryV2Module)
+      .overrideModule(NotificationsModuleV2)
+      .useModule(TestNotificationsModuleV2)
       .compile();
 
     const configurationService = moduleFixture.get<IConfigurationService>(
       IConfigurationService,
     );
+    notificationServiceV2 = moduleFixture.get(NotificationsServiceV2);
     safeConfigUrl = configurationService.getOrThrow('safeConfig.baseUri');
     networkService = moduleFixture.get(NetworkService);
 
     app = await new TestAppProvider().provide(moduleFixture);
     await app.init();
   });
 
-  const buildInputDto = (): RegisterDeviceDto =>
-    registerDeviceDtoBuilder()
-      .with(
-        'safeRegistrations',
-        Array.from({ length: 4 }).map((_, i) => {
-          return safeRegistrationBuilder()
-            .with('chainId', i.toString())
-            .build();
-        }),
-      )
+  const buildInputDto = async (
+    safeRegistrationsLength: number = 4,
+  ): Promise<RegisterDeviceDto> => {
+    const uuid = faker.string.uuid() as UUID;
+    const cloudMessagingToken = faker.string.uuid() as UUID;
+    const timestamp = faker.date.recent();
+    timestamp.setMilliseconds(0);
+    const timestampWithoutMilliseconds = timestamp.getTime();
+
+    const safeRegistrations = await Promise.all(
+      Array.from({ length: safeRegistrationsLength }).map(async () => {
+        const safeRegistration = await safeRegistrationBuilder({
+          signaturePrefix: 'gnosis-safe',
+          uuid,
+          cloudMessagingToken,
+          timestamp: timestampWithoutMilliseconds,
+        });
+        return safeRegistration
+          .with('chainId', faker.number.int({ min: 1, max: 100 }).toString())
+          .build();
+      }),
+    );
+
+    return (
+      await registerDeviceDtoBuilder({
+        uuid,
+        cloudMessagingToken,
+        timestamp: timestampWithoutMilliseconds,
+      })
+    )
+      .with('safeRegistrations', safeRegistrations)
       .build();
+  };
 
   const rejectForUrl = (url: string): Promise<never> =>
     Promise.reject(`No matching rule for url: ${url}`);
 
   describe('POST /register/notifications', () => {
-    it('Success', async () => {
-      const registerDeviceDto = buildInputDto();
-      networkService.get.mockImplementation(({ url }) =>
-        url.includes(`${safeConfigUrl}/api/v1/chains/`)
-          ? Promise.resolve({ data: chainBuilder().build(), status: 200 })
-          : rejectForUrl(url),
-      );
-      networkService.post.mockImplementation(({ url }) =>
-        url.includes('/api/v1/notifications/devices/')
-          ? Promise.resolve({ data: {}, status: 200 })
-          : rejectForUrl(url),
-      );
+    it.each([5, 20, 100])(
```

Comment:
Nit: we could just use a random `faker.number.int` and have the one test to save running the registration so many times. What do you think?

Reply:
As discussed, it’s better not to run the test a random number of times. I’ve removed the 100 iterations, as 5 and 20 are sufficient.